byzcoin/service.go

```diff
@@ -1245,6 +1242,21 @@ func (s *Service) getTxs(leader *network.ServerIdentity, roster *onet.Roster, sc
 	return s.txBuffer.take(string(scID))
 }
 
+func (s *Service) loadNonceFromTxs(txs TxResults) ([]byte, error) {
+	if len(txs) == 0 {
+		return nil, errors.New("no transactions")
+	}
+	instrs := txs[0].ClientTransaction.Instructions
```

Comment:
Two more checks to be even more conservative: if len(instrs) != 1 { return nil, fmt.Errorf("expected 1 instruction, got %v", len(instrs)) } and if instrs[0].Spawn == nil { return nil, errors.New("first instruction is not a Spawn") }